lib/ansible/executor/task_executor.py

```diff
@@ -819,8 +819,16 @@ def _get_connection(self, variables, templar):
             raise AnsibleError("the connection plugin '%s' was not found" % conn_type)
 
         # load become plugin if needed
-        if boolean(cvars.get('ansible_become', self._task.become)):
-            become_plugin = self._get_become(cvars.get('ansible_become_method', self._task.become_method))
+        if cvars.get('ansible_become') is not None:
+            become = boolean(templar.template(cvars.get('ansible_become')))
+        else:
+            become = self._task.become
+
+        if become:
+            if cvars.get('ansible_become_method'):
+                become_plugin = self._get_become(templar.template(cvars.get('ansible_become_method')))
+            else:
+                become_plugin = self._get_become(self._task.become_method)
```

Comment:
might also need 'ansible_connection' above

Reply:
I hoped it did, but it doesn't. Fixed, and added a test for these.